content/learn/book/the-game-loop/time-and-timers.md

```diff
@@ -0,0 +1,277 @@
+
+[`Time`]: https://docs.rs/bevy/latest/bevy/prelude/struct.Time.html
+[`Instant::now()`]: https://doc.rust-lang.org/std/time/struct.Instant.html#method.now
+[`TimeSystem`]: https://docs.rs/bevy/latest/bevy/time/struct.TimeSystem.html
+[`First`]: https://docs.rs/bevy/latest/bevy/app/struct.First.html
+
+## Frame-rate independence and delta time
+
+Suppose we want to move our player to the right:
+
+```rust
+use bevy::prelude::*;
+
+#[derive(Component)]
+struct Player;
+
+fn move_player(mut player_transform: Single<&mut Transform, With<Player>>){
+    const PLAYER_MOVEMENT: f32 = 1.;
+    
+    player_transform.translation.x += PLAYER_MOVEMENT;
+}
+```
+
+Every frame, our player will move 1 unit to the right. Great!
+But what happens when our game stutters, and the frame rate drops?
+Suddenly, rather than moving 60 units per second at our target 60 frames per second,
+our player is moving at an unsteady 20-30 units per second. Oh no!
+
+Instead, we can compensate for this effect by fixing our *speed* (or other rates of change per second),
+and then multiplying by the elapsed time.
+
+```rust, hide_lines=1-4
+# use bevy::prelude::*;
+# 
+# #[derive(Component)]
+# struct Player;
+#
+fn move_player(mut player_transform: Single<&mut Transform, With<Player>>, time: Res<Time>){
+    // At 60 FPS, this will be the same as before
+    const PLAYER_SPEED: f32 = 60.;
+    
+    player_transform.translation.x += PLAYER_SPEED * time.delta_secs_f32();
+}
+```
+
+This technique is commonly known as ["delta time"] among game devs, because physicists use "delta" to mean "a change in a quantity".
+
+["delta time"]: https://en.wikipedia.org/wiki/Delta_timing
+
+## Pausing and time control
+
+With all of our time-dependent logic driven by the delta time,
+we can start playing tricks with the value of [`Time`] to implement features like pausing and slowing down the game.
+To do this, we need to change how we account for [`Virtual`] (in-game) time:
+
+```rust
+use bevy::prelude::*;
+
+fn toggle_pause(mut time: ResMut<Time<Virtual>>) {
+    if time.is_paused() {
+        time.unpause();
+    } else {
+        time.pause();
+    }
+}
+
+#[derive(Event)]
+struct SetGameSpeed(f32);
+
+fn set_game_speed(mut time: ResMut<Time<Virtual>>, events: EventReader::<SetGameSpeed>){
+  if let Some(new_speed) = events.iter().last(){
+      time.set_relative_speed(new_speed.0);
+  }
+}
+```
+
+If your systems uniformly rely on [`Time`], this will effect your entire game:
+halting, slowing or speeding up animations, movement, projectiles, physics and so on.
+Alternatively, [states] and [run conditions] can be used to skip systems while the game is paused.
+
+[states]: ../control-flow/states.md
+[run conditions]: ../control-flow/run-conditions.md
+
+## Fixing your timestep
+
+Compensating for fluctating frame times using delta time is a great start.
+But as the timeless [fix your timestep article] explains for projects that require a higher level of stability and reproducibility,
```

Comment:
this is a stylistic nit, but I think as a reader I am used to book names, articles and so on that are referred to by name to be written as the title is:


```suggestion
But as the timeless [*Fix Your Timestep!*] article explains for projects that require a higher level of stability and reproducibility,
```

Reply:
Personally, I'd also explicitly credit the author when referencing their article in a public book-like context like this

```suggestion
But as the timeless [*Fix Your Timestep!*] article by Glenn Fiedler explains, for projects that require a higher level of stability and reproducibility,
```